src/mm-input-mask/mm-input-mask.js

```diff
@@ -0,0 +1,396 @@
+							auto:auto,
+							type:_types.GROUP, 
+							style:style,
+							loaded: false,
+							contentWidth:this.contentWidthHandler
+						};
+						groups.push(o);
+						return o;
+
+					case "SEP":
+						var chars = node.getAttribute("chars");
+						seps.push(chars);
+						this.sepsLen += chars.length;
+						return {
+							type:_types.SEP, 
+							value:chars, 
+							style:{
+								sep:true, 
+								placeholder:true
+							}
+						};
+				}
+			}, this);
+			this.seps = seps;
+			this.groups = groups;
+
+		},
+
+		contentWidthHandler: function(style, val, placeholder, loaded) {
+			var w = 0.0;
+			var check = val || placeholder || "";
+			while(check.length < this.max) {
+				check += "S";
+			}
+			if (!val && placeholder) {
+				w = Measure.textWidth(null, check, "italic 13px Arimo");
+			} else {
+				w = Measure.textWidth(null, check, "13px Arimo");
+			}
+			style.width = w + "px";
+			return style;
+		},
+
+		placeholderChanged: function(oldPlaceholder, newPlaceholder) {
+			this.chunkValue(this.placeholder, "placeholder");
+		},
+
+		valueChanged: function(oldValue, newValue) {
+			if (newValue) {
+				var placeholder = newValue.length <= this.sepsLen;
+				this.maskConfig.filter(function(m) { return m.type === _types.SEP}).forEach(function(sep) {
+					sep.style = {sep:true, placeholder:placeholder};
+				},this);
+			}
+			if (this.ignoreInternal) {
+				this.ignoreInternal = false;
+				return;
+			}
+			this.chunkValue(newValue);
+		},
+
+		chunkValue: function(value, type) {
+			if (!type) type = "value";
+			//clear case
+			if (value === "" || value === null) {
+				for(var i=0; i<this.groups.length; i++) {
+					var g = this.groups[i];
+					this.maskConfig[g.index][type] = "";
+				}
+			//chunk our values and assign them to sub fields
+			} else {
+				var sepReg = "[" +  this.seps.join("")+ "]+",
+					srcPath = Path.get("restrict.source"),
+					base = this.maskConfig.map(function(item) {
+						switch(item.type) {
+							case _types.GROUP:
+								item = this.groups.filter(function(g) {
+									return g.index === item.index;
+								})[0];
+								var src = srcPath.getValueFrom(item) || ".*";
+								if (type === "placeholder") {
+									src = ".*";
+								}
+								if (item.min !== item.max) {
+									return "(" + src + ")";
+								} else {
+									src = src.replace("+","");
+									src = src.replace("*","");
+									return "(" + src + "{" + item.min + "})";
+								}
+								break;
+							case _types.SEP:
+								return sepReg;
+						}
+					},this),
+					r = new RegExp(base.join(""), "g"),
+					res = r.exec(value);
+
+				if (res) {
+					res.shift();
+
+					for(var i=0; i<this.groups.length; i++) {
+						var g = this.groups[i];
+						this.maskConfig[g.index][type] = res[i];
+					}
+				}
+			}
+		},
+
+		updateGroups: function() {
+			this.groupSel = this.groups.map(function(o) {
+				return this.shadowRoot.querySelector("#mini"+o.index);
+			}.bind(this));
```

Comment:
A premature "optimization" ... instead of binding, you could pass "this" as the second argument to this.groups.map() -- and then you could extract the transformation function (since it doesn't close around any variables) so it doesn't get recreated at every invocation of updateGroups()